arlo-client/src/components/Atoms/Table.tsx

```diff
@@ -0,0 +1,145 @@
+    border-spacing: 0;
+    color: #394b59; /* BlueprintJS dark-gray5 */
+  }
+
+  th,
+  td {
+    margin: 0;
+    padding: 0.5rem;
+    text-align: left;
+  }
+
+  tr:nth-child(even) {
+    background-color: #f5f8fa; /* BlueprintJS light-gray5 */
+  }
+`
+
+interface ITableProps<T extends object> {
+  data: T[]
+  columns: Column<T>[]
+}
+
+const FilterInput = <T extends object>({
+  column: { filterValue, setFilter },
+}: {
+  column: ColumnInstance<T>
+}) => (
+  <FilterWrapper>
+    <div className="bp3-input-group .modifier">
+      <span className="bp3-icon bp3-icon-filter"></span>
+      <input
+        type="text"
+        className="bp3-input"
+        placeholder="Filter by jurisdiction name..."
+        value={filterValue || ''}
+        onChange={e => setFilter(e.target.value || undefined)}
+      />
+    </div>
+  </FilterWrapper>
+)
+
+const Table = <T extends object>({ data, columns }: ITableProps<T>) => {
+  const {
+    getTableProps,
+    getTableBodyProps,
+    headers,
+    rows,
+    prepareRow,
+  } = useTable(
+    {
+      data: React.useMemo(() => data, [data]),
+      columns: React.useMemo(() => columns, [columns]),
+      defaultColumn: React.useMemo(() => ({ Filter: FilterInput }), []),
+    },
+    useFilters,
+    useSortBy
+  )
+
+  /* eslint-disable react/jsx-key */
+  /* All the keys are added automatically by react-table */
+
+  const filterableColumns = headers.filter(column => column.filter)
+  if (filterableColumns.length > 1)
+    throw Error('Only allowed to have one filterable column max')
+  const [filterColumn] = filterableColumns
+  console.log(headers, headers[0].canFilter, filterColumn)
+
+  return (
+    <Wrapper>
+      {filterColumn && filterColumn.render('Filter')}
+      <StyledTable {...getTableProps()}>
+        <thead>
+          <tr>
+            {headers.map(column => (
+              <th {...column.getHeaderProps(column.getSortByToggleProps())}>
+                {column.render('Header')}
+                <span>
+                  {column.isSorted ? (
+                    column.isSortedDesc ? (
+                      <Icon icon="caret-down" />
+                    ) : (
+                      <Icon icon="caret-up" />
+                    )
+                  ) : (
+                    <Icon icon="double-caret-vertical" />
+                  )}
+                </span>
+              </th>
+            ))}
+          </tr>
+        </thead>
+        <tbody {...getTableBodyProps()}>
+          {rows.map(row => {
+            prepareRow(row)
+            return (
+              <tr {...row.getRowProps()}>
+                {row.cells.map(cell => (
+                  <td {...cell.getCellProps()}>{cell.render('Cell')}</td>
+                ))}
+              </tr>
+            )
+          })}
+        </tbody>
+      </StyledTable>
+    </Wrapper>
+  )
+}
+
+export const sortByRank = <T extends object>(rank: (data: T) => number) =>
+  // eslint-disable-next-line react-hooks/rules-of-hooks
```

Comment:
@jonahkagan Maybe document here why this is disabled? I'm presuming it's because you need to memoize the function passed to the sortBy feature, and the linter complains because this function is technically not a hook or a functional component?

Reply:
Yeah I think you nailed it.